esmf_regrid/_esmf_sdo.py

```diff
@@ -90,14 +98,14 @@ class GridInfo(SDO):
     This class holds information about lat-lon type grids. That is, grids
     defined by lists of latitude and longitude values for points/bounds
     (with respect to some coordinate reference system i.e. rotated pole).
-    It contains methods for translating this information into :mod:`ESMF` objects.
+    It contains methods for translating this information into :mod:`esmpy` objects.
```

Comment:
I don't think your changes to Sphinx domains will work, since those are based on this line:

https://github.com/SciTools-incubator/iris-esmf-regrid/blob/f0781aaff0c657ea878173f6fdcde64dfd113be6/docs/src/conf.py#L118